frontend/src/features/admin-form/responses/FeedbackPage/FeedbackDownloadButton.tsx

```diff
@@ -1,39 +1,36 @@
-import { useMemo } from 'react'
-import { CSVLink } from 'react-csv'
+import { useCallback } from 'react'
 import { BiDownload } from 'react-icons/bi'
 
-import { ProcessedFeedbackMeta } from '~shared/types'
-
 import Button from '~components/Button'
 
+import { useFormFeedbackMutations } from '../../common/mutations'
+
 type FeedbackDownloadButtonProps = {
   isDisabled: boolean
   formId: string | undefined
-  feedback: ProcessedFeedbackMeta[] | undefined
+  formTitle: string | undefined
 }
 
 export const FeedbackDownloadButton = ({
-  feedback,
   isDisabled,
   formId,
+  formTitle,
 }: FeedbackDownloadButtonProps) => {
-  const data = useMemo(() => {
-    if (!feedback) return ''
-    return feedback?.map((entry) => ({
-      index: entry.index,
-      date: entry.date,
-      feedback: entry.comment,
-      rating: entry.rating,
-    }))
-  }, [feedback])
+  const { downloadFormFeedbackMutation } = useFormFeedbackMutations(
+    formId ?? '',
+    formTitle ?? '',
+  )
+
+  const handleClick = useCallback(
+    () => downloadFormFeedbackMutation.mutate(),
+    [downloadFormFeedbackMutation],
+  )
 
   return (
     <Button
-      disabled={isDisabled}
-      as={CSVLink}
-      filename={`${formId}-feedback.csv`}
-      data={data}
-      target="_blank"
+      disabled={isDisabled || downloadFormFeedbackMutation.isLoading}
```

Comment:
should we use the `isDisabled` prop instead? that chakra prop injects a bunch of a11y props to denote a disabled state.